third_party/openocd/reset_on_dmi_op_error.patch

```diff
@@ -0,0 +1,60 @@
+diff --git a/src/target/riscv/riscv-013.c b/src/target/riscv/riscv-013.c
+index 4e6c8dc36..94d178d06 100644
+--- a/src/target/riscv/riscv-013.c
++++ b/src/target/riscv/riscv-013.c
+@@ -538,6 +538,33 @@ static dmi_status_t dmi_scan(struct target *target, uint32_t *address_in,
+ 	return buf_get_u32(in, DTM_DMI_OP_OFFSET, DTM_DMI_OP_LENGTH);
+ }
+ 
++/**
++ * The OpenTitan rv_dm can get stuck in certain circumstances. This can easily
++ * be detected because tdo is stuck at one so reading DTMCS returns all ones,
++ * a clearly invalid value. Also the rv_dm never sends back an operation status
++ * other than 0 (success) or 3 (busy). When this happens, continue reading until it gets
++ * unstuck and performs a hard dmi reset.
++ */
++static int ott_recover(struct target *target)
++{
++	/* wait until the dm is unstuck */
++	int tries = 100;
++	while(1) {
++		uint32_t dtmcontrol = dtmcontrol_scan(target, DTM_DTMCS_DMIRESET);
++		if (dtmcontrol == 0xffffffff) {
++			LOG_ERROR("OTT rv_dm is stuck, wait until it is live again");
++		}
++		else
++			break;
```

Comment:
Suggest to put on one line (`else break;`) or add parentheses

Reply:
I am trying to follow the OpenOCD source code convention of that file (which is a bit flexible it seems). I am happy to change to OpenTitan's convention but I am not sure if this is the right approach.

Reply:
I have updated the patch.